docs/native-modules-android.md

```diff
@@ -119,236 +132,420 @@ import java.util.ArrayList;
+
 The JavaScript counterpart of this method returns a Promise. This means you can use the `await` keyword within an async function to call it and wait for its result:
 
 ```jsx
-const measureLayout = async () => {
+const onSubmit = async () => {
   try {
-    var {
-      relativeX,
-      relativeY,
-      width,
-      height
-    } = await UIManager.measureLayout(100, 100);
-
-    console.log(
-      relativeX + ':' + relativeY + ':' + width + ':' + height
+    const eventId = await CalendarModule.createCalendarEvent(
+      'Party',
+      'My House'
     );
+    console.log(`Created a new event with id ${eventId}`);
   } catch (e) {
     console.error(e);
   }
 };
+```
+
+The reject method takes different combinations of the following arguments:
+
+```java
+String code, String message, WritableMap userInfo, Throwable throwable
+```
+
+For more detail, you can find the `Promise.java` interface [here](https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/bridge/Promise.java#L1). If `userInfo` is not provided, ReactNative will set it to null. For the rest of the parameters React Native will use a default value. The `message` argument provides the error `message` shown at the top of an error call stack. Below is an example of the error message shown in Javascript from the following reject call in Java.
 
-measureLayout();
+Java reject call:
+
+```java
+promise.reject("Create Event error", "Error parsing date", e);
 ```
 
-### Threading
+Error message in React Native App when promise is rejected:
 
-Native modules should not have any assumptions about what thread they are being called on, as the current assignment is subject to change in the future. If a blocking call is required, the heavy work should be dispatched to an internally managed worker thread, and any callbacks distributed from there.
+<figure>
+  <img src="/docs/assets/native-modules-android-errorscreen.png" width="200" alt="Image of error message in React Native app." />
+  <figcaption>Image of error message</figcaption>
+</figure>
 
-### Sending Events to JavaScript
+### Sending Events to Javascript
 
-Native modules can signal events to JavaScript without being invoked directly. The easiest way to do this is to use the `RCTDeviceEventEmitter` which can be obtained from the `ReactContext` as in the code snippet below.
+Native modules can signal events to JavaScript without being invoked directly. For example, you might want to signal to Javascript a reminder that a calendar event from the native Android calendar app will occur soon. The easiest way to do this is to use the `RCTDeviceEventEmitter` which can be obtained from the `ReactContext` as in the code snippet below.
 
 ```java
+
```

Comment:
Please clean up the unnecessary new line/whitespaces at the beginning and at the end of the code blocks.

Reply:
done